docs/tutorials/jax-primitives.md

```diff
@@ -0,0 +1,592 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.15.2
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(jax-internals-jax-primitives)=
+# JAX internals 301: JAX primitives
+
+This tutorial explains the interface that a JAX primitive must support to allow JAX to perform all its transformations. JAX implements certain composable transformations of Python functions, such as {func}`jax.jit`, {func}`jax.grad` and {func}`jax.vmap`, in a *JAX-traceable* way. This means that when a Python function is executed, the only operations it applies to the data are either:
```

Comment:
Have we defined anywhere what a "primitive" is? If not, we should do that here. If so, we should link to it.

Reply:
Will rephrase the original, since it was not 100% clear.

Reply:
:+1: :+1: Thought about this too